src/pgduckdb_types.cpp

```diff
@@ -30,94 +30,72 @@ extern "C" {
 
 namespace pgduckdb {
 
-struct BoolArray {
-public:
-	static ArrayType *
-	ConstructArray(Datum *datums, bool *nulls, int ndims, int *dims, int *lower_bound) {
-		return construct_md_array(datums, nulls, ndims, dims, lower_bound, BOOLOID, sizeof(bool), true, 'c');
-	}
-	static duckdb::LogicalTypeId
-	ExpectedType() {
-		return duckdb::LogicalTypeId::BOOLEAN;
-	}
-	static Datum
-	ConvertToPostgres(const duckdb::Value &val) {
-		return Datum(val.GetValue<bool>());
-	}
-};
+static Datum
+ConvertBoolDatum(const duckdb::Value &value) {
+	return value.GetValue<bool>();
+}
 
-template <int32_t OID>
-struct PostgresIntegerOIDMapping {};
+static Datum
+ConvertCharDatum(const duckdb::Value &value) {
```

Comment:
I very much like these new functions. I think taking that refactor on its own would make the code already a lot better.